MovieApp/DataLayer/Repositories/MovieRepository/MovieRepository.swift

```diff
@@ -113,4 +121,42 @@ class MovieRepository: MovieRepositoryProtocol {
             .eraseToAnyPublisher()
     }
 
+    private func bindSources() {
```

Comment:
Although this works, this will update the data only once on app load, am I correct?
So, either this is exactly what you wanted or we want the following:
* always try to fetch from the network
* if it succeeds, save in DB
* if it fails, read from DB

You can do all of that in one chain, use `handleEvents` & `*Error` operators.

Reply:
Yes, you are correct. That was what I wanted since I figured that the data in the app isn't very dynamic so fetching it remotely once per launch is good enough and would be more efficient since it results in less network calls.